Ignore `test_version` on version bump PR

When following the [release process](https://deltares.github.io/Ribasim/dev/release.html) and creating #1654, it is impossible to bump the version numbers before we tag the new version and have the `test_version` in `test_cli.py` pass.

```
build.tests.test_cli.test_version
16:54:35       AssertionError: assert '2024.10.0' in 'ribasim 2024.9.0-57-g2e9e1db-dirty\n'
       +  where '2024.10.0' = ribasim.__version__
       +  and   'ribasim 2024.9.0-57-g2e9e1db-dirty\n' = CompletedProcess(args=[PosixPath('/opt/teamcityagent/work/ecd2b8f9b25b1609/ribasim/build/ribasim/ribasim'), '--version'], returncode=0, stdout='ribasim 2024.9.0-57-g2e9e1db-dirty\n', stderr='').stdout
16:54:35       def test_version():
              result = subprocess.run(
                  [executable, "--version"], check=True, capture_output=True, text=True
              )
          
      >       assert ribasim.__version__ in result.stdout
      E       AssertionError: assert '2024.10.0' in 'ribasim 2024.9.0-57-g2e9e1db-dirty\n'
      E        +  where '2024.10.0' = ribasim.__version__
      E        +  and   'ribasim 2024.9.0-57-g2e9e1db-dirty\n' = CompletedProcess(args=[PosixPath('/opt/teamcityagent/work/ecd2b8f9b25b1609/ribasim/build/ribasim/ribasim'), '--version'], returncode=0, stdout='ribasim 2024.9.0-57-g2e9e1db-dirty\n', stderr='').stdout
      
      build/tests/test_cli.py:33: AssertionError
```

We can ignore this error since the result of `ribasim --version` is picked up from the git tag. But it would be nice not to have failing tests on the PRs where it is essential that everything is ok.

This feature was added in #1479.

--- a/build/build.jl
+++ b/build/build.jl
@@ -39,16 +39,15 @@ Usage: `ribasim path/to/model/ribasim.toml`
 Documentation: https://ribasim.org/
 """
 
-function set_version(filename, version; group = nothing)
+"Use the git tag for `ribasim --version`,
+so dev builds can be identified by <tag>-g<short-commit>"
+function set_version(filename::String, tag::String)::Nothing
     data = TOML.parsefile(filename)
-    if !isnothing(group)
-        data[group]["version"] = version
-    else
-        data["version"] = version
-    end
+    data["package"]["version"] = tag
     open(filename, "w") do io
         TOML.print(io, data)
     end
+    return nothing
 end
 
 """
@@ -119,7 +118,7 @@ function add_metadata(project_dir, license_file, output_dir, git_repo, readme)
     end
 
     # Override the Cargo.toml file with the git version
-    set_version("cli/Cargo.toml", tag; group = "package")
+    set_version("cli/Cargo.toml", tag)
 end
 
 main()

--- a/build/build.jl
+++ b/build/build.jl
@@ -82,15 +82,6 @@ function add_metadata(project_dir, license_file, output_dir, git_repo, readme)
         force = true,
     )
 
-    # since the exact Ribasim version may be hard to find in the Manifest.toml file
-    # we can also extract that information, and add it to the README.md
-    manifest = TOML.parsefile(normpath(git_repo, "Manifest.toml"))
-    if !haskey(manifest, "manifest_format")
-        error("Manifest.toml is in the old format, run Pkg.upgrade_manifest()")
-    end
-    julia_version = manifest["julia_version"]
-    ribasim_entry = only(manifest["deps"]["Ribasim"])
-    version = ribasim_entry["version"]
     repo = GitRepo(git_repo)
     branch = LibGit2.head(repo)
     commit = LibGit2.peel(LibGit2.GitCommit, branch)
@@ -120,11 +111,9 @@ function add_metadata(project_dir, license_file, output_dir, git_repo, readme)
         This build uses the Ribasim version mentioned below.
 
         ```toml
-        release = "$tag"
+        version = "$tag"
         commit = "$url/$short_commit"
         branch = "$url/$short_name"
-        julia_version = "$julia_version"
-        core_version = "$version"
         ```"""
         println(io, version_info)
     end

--- a/build/tests/test_cli.py
+++ b/build/tests/test_cli.py
@@ -33,7 +33,7 @@ def test_version():
 
     # ribasim --version is based on the git tag so can be different from
     # ribasim.__version__ during development
-    version_pattern = r"ribasim \d{4,}\.\d{1,}\.\d{1,}"
+    version_pattern = r"ribasim \d{4,}\.\d+\.\d+"
     assert re.match(version_pattern, result.stdout)
 
 

--- a/build/tests/test_cli.py
+++ b/build/tests/test_cli.py
@@ -1,3 +1,4 @@
+import re
 import subprocess
 from pathlib import Path
 
@@ -30,7 +31,10 @@ def test_version():
         [executable, "--version"], check=True, capture_output=True, text=True
     )
 
-    assert ribasim.__version__ in result.stdout
+    # ribasim --version is based on the git tag so can be different from
+    # ribasim.__version__ during development
+    version_pattern = r"ribasim \d{4,}\.\d{1,}\.\d{1,}"
+    assert re.match(version_pattern, result.stdout)
 
 
 def test_help():

